### PR TITLE
Add TextInfo casing tests

### PR DESCRIPTION
**Pull request by ellismg**
An initial set of TextInfo casing tests. I looked at the scinerios we
have coverage for in ToF, but wrote the actual tests from scratch in
idomatic XUnit style.

In addition, I have added some specific tests for the Turkish I casing,
since the characters have special casing behavior in Turkish and some
other locales as well as test cases to ensure we have the simple casing
in places where the Unicode standard normally defines special cases
which are either context sensitive or would change the number of
codepoints in a cased string.

### CODE REVIEW

File: src/System.Globalization/tests/TextInfo.cs [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+        Assert.Equal("THIS IS A LONGER TEST CASE", ti.ToUpper("this is a longer test case"));
+        Assert.Equal("THIS IS A LONGER MIXED CASE TEST CASE", ti.ToUpper("this Is A LONGER mIXEd casE test case"));
+
+        Assert.Equal("THIS \t HAS \t SOME \t TABS", ti.ToUpper("this \t HaS \t somE \t TABS"));
+
+        Assert.Equal('1', ti.ToUpper('1'));
+        Assert.Equal("123", ti.ToUpper("123"));
+
+        Assert.Equal("EMBEDED\0NULL\0BYTE\0", ti.ToUpper("embeded\0NuLL\0Byte\0"));
```
Comment (stephentoub) [inline src/System.Globalization/tests/TextInfo.cs]: Extreme nit: embedded

File: src/System.Globalization/tests/TextInfo.cs [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+        Assert.Equal('\u00D3', ti.ToUpper('\u00F3'));
+        Assert.Equal("\u00D3", ti.ToUpper("\u00F3"));
+
+        Assert.Equal('\u2603', ti.ToUpper('\u2603'));
```
Comment (stephentoub) [inline src/System.Globalization/tests/TextInfo.cs]: For these specific unicode character tests, is it worth adding a comment indicating what it is / why it's chosen?

Comment (ellismg) [inline src/System.Globalization/tests/TextInfo.cs]: Can do (the first is a character that actually cases (an o with an accent mark), the second is a character that does not (a snowman)).

File: src/System.Globalization/tests/TextInfo.cs (hunk 1 above)
Comment (ellismg) [inline src/System.Globalization/tests/TextInfo.cs]: Fixed.

File: src/System.Globalization/tests/System.Globalization.Tests.csproj [hunk 3]
```diff
@@ -0,0 +1,25 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (stephentoub) [inline src/System.Globalization/tests/System.Globalization.Tests.csproj]: Could you add a .sln with this as well?

**Comment by stephentoub** [comment]
A few minor comments, but otherwise LGTM. Thanks!

File: src/System.Globalization/tests/TextInfo.cs (hunk 2 above)
Comment (ellismg) [inline src/System.Globalization/tests/TextInfo.cs]: I also added a test for a character outside the BMP that does have a casepair.

File: src/System.Globalization/tests/System.Globalization.Tests.csproj (hunk 3 above)
Comment (ellismg) [inline src/System.Globalization/tests/System.Globalization.Tests.csproj]: Can do.

**Comment by ellismg** [comment]
Thanks Steve, I addressed all your feedback, hopefully fixed the issue that was causing the change to fail in CI (which I didn't hit locally because I am on Windows 10) and squashed.

**Comment by stephentoub** [comment]
Thanks, Matt.  Assuming it passes, LGTM.

**Pull request merged**